File: samples/TapdaqExamples/InterstitialOnBootup/ViewController.m
//
//  ViewController.m
//  InterstitialOnBootup
//
//  Created by Nick Reffitt on 22/12/2016.
//  Copyright © 2016 Tapdaq. All rights reserved.
//

#import "ViewController.h"
#import <Tapdaq/Tapdaq.h>
@interface ViewController ()

@end

@implementation ViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    [[NSNotificationCenter defaultCenter] addObserver:self selector:@selector(receivedLog:) name:@"TapdaqLogMessageNotification" object:nil];
}

- (void)dealloc {
    [[NSNotificationCenter defaultCenter] removeObserver:self];
}


- (void)didReceiveMemoryWarning {
    [super didReceiveMemoryWarning];
}

- (void)receivedLog:(NSNotification *)notification {
    NSString *text = [self.textView text];
    NSString *message = notification.userInfo[@"message"];
    NSString *textAddition = [NSString stringWithFormat:@"%@\n\n%@", message, text];
    [self.textView setText:textAddition];
}

- (IBAction)showDebugger:(id)sender {
    [[Tapdaq sharedSession] presentDebugViewController];
}


@end

File: TapdaqExamples/InterstitialOnBootupSwift/ViewController.swift
//
//  ViewController.swift
//  InterstitialOnBootupSwift
//
//  Created by Nick Reffitt on 16/05/2017.
//  Copyright © 2017 Tapdaq. All rights reserved.
//

import UIKit
import Tapdaq

class ViewController: UIViewController, TapdaqDelegate {
    
    @IBOutlet var textView: UITextView!

    override func viewDidLoad() {
        super.viewDidLoad()
        NotificationCenter.default.addObserver(self, selector: #selector(receivedLog(_:)), name: NSNotification.Name("TapdaqLogMessageNotification"), object: nil)
    }

    deinit {
        NotificationCenter.default.removeObserver(self)
    }
    
    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
        // Dispose of any resources that can be recreated.
    }
    
    @objc private func receivedLog(_ notification: NSNotification) {
        guard let message = notification.userInfo?["message"] as? String else {
            return
        }
        
        let text: String = textView.text
        let textAddition: String = message + "\n\n" + text
        textView.text = textAddition
    }
    
    @IBAction func showDebugger(_ sender: Any?) {
        Tapdaq.sharedSession().presentDebugViewController()
    }

    
}

